.github/workflows/test.yaml

```diff
@@ -129,6 +129,7 @@ jobs:
           - python-version: '3.11'
             runs-on: 'macos-12'
             job-name: 'macos-12-default-tests'
+            tox-env: 'macos-withcov-alldeps'
```

Comment:
with this change, we make sure that cfreactor deps are installed and we don't skip the cfreactor targeted unit/integration tests, while all the other tests are executed with the select reactor.

If we care about cfreactor, I think is best to run the full test suite with the cfreator...
Most probably it will not detect any other failure, but I think is best to have it and increase the probability of detecting regressions or cfreator-related issues.

and I guess that we still want to run select reactor on macOS

So maybe have these CI macOS environments:

* Python 3.7 with select reactor on macOS 12 - the minimum python that we support
* Python 3.11 with cfreactor reactor on macOS 12 - the maximum/latest python that we support

Reply:
I don't necessarily object but I'm also not going to volunteer for this work.  As-is, this PR adds a lot of CFReactor coverage on CI (it was 0 before, now it's "a lot").  Since you mentioned below that you don't think this is a blocker, I will leave it for another time.

Reply:
For reasons related to the way global state works in the corefoundation API, this would be *extremely* difficult to implement without completely rewriting trial from scratch. So I am not sure we should even bother tackling this later.

Reply:
(In any case, definitely nothing to do on this PR so resolving)